source/backend/api/Areas/Leases/Controllers/LeaseController.cs

```diff
@@ -234,6 +236,17 @@ public IActionResult GetAcquisitionTeamMembers()
             return new JsonResult(_mapper.Map<IEnumerable<LeaseFileTeamModel>>(team));
         }
 
+        [HttpGet("{id:long}/historical")]
+        [Produces("application/json")]
+        [HasPermission(Permissions.LeaseView)]
+        [ProducesResponseType(typeof(LeaseModel), 200)]
+        [SwaggerOperation(Tags = new[] { "lease" })]
+        [TypeFilter(typeof(NullJsonResultFilter))]
+        public IActionResult GetLeaseAtTime([FromRoute] long id, [FromQuery] DateTime time)
```

Comment:
Hmm, why does this controller directly access the repository instead of the service?

Reply:
After adding them to the service I think if the service is not providing any additional logic, the controller could directly call the repository.